common-ui-compose/src/main/java/app/tivi/common/compose/Insets.kt

```diff
@@ -0,0 +1,300 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+@file:Suppress("NOTHING_TO_INLINE")
+
+package app.tivi.common.compose
+
+import androidx.compose.Composable
+import androidx.compose.Providers
+import androidx.compose.Stable
+import androidx.compose.StructurallyEqual
+import androidx.compose.getValue
+import androidx.compose.mutableStateOf
+import androidx.compose.onCommit
+import androidx.compose.remember
+import androidx.compose.setValue
+import androidx.compose.staticAmbientOf
+import androidx.core.view.ViewCompat
+import androidx.core.view.doOnAttach
+import androidx.ui.core.DensityAmbient
+import androidx.ui.core.Modifier
+import androidx.ui.core.ViewAmbient
+import androidx.ui.core.composed
+import androidx.ui.layout.absolutePadding
+import androidx.ui.layout.height
+import androidx.ui.layout.width
+import androidx.ui.unit.Density
+import androidx.ui.unit.dp
+
+/**
+ * Main holder of our inset values.
+ */
+@Stable
+class DisplayInsets {
+    val systemBars by lazy(LazyThreadSafetyMode.NONE) { Insets() }
+}
+
+@Stable
+class Insets {
+    var left by mutableStateOf(0.dp, StructurallyEqual)
+        internal set
+    var top by mutableStateOf(0.dp, StructurallyEqual)
+        internal set
+    var right by mutableStateOf(0.dp, StructurallyEqual)
+        internal set
+    var bottom by mutableStateOf(0.dp, StructurallyEqual)
+        internal set
+
+    /**
+     * TODO: doesn't currently work
+     */
+    var visible by mutableStateOf(true)
+        internal set
+
+    override fun equals(other: Any?): Boolean {
```

Comment:
Are the equals/hashCode implementations still needed or are these leftover from a version where this was immutable?

Reply:
Nah, I can remove them